src/modules/UI/components/ErrorAlert/actions.js

```diff
@@ -1,13 +1,25 @@
 // @flow
+import { Alert } from 'react-native'
+
+import * as actions from '../../../../actions/indexActions.js'
+import s from '../../../../locales/strings.js'
+import type { Dispatch } from '../../../../modules/ReduxTypes'
 
 const PREFIX = 'UI/components/ErrorAlert/'
 export const DISPLAY_ERROR_ALERT = PREFIX + 'DISPLAY_ERROR_ALERT'
 export const DISMISS_ERROR_ALERT = PREFIX + 'DISMISS_ERROR_ALERT'
 
-export const displayErrorAlert = (message: string) => ({
+export const displayErrorAlertStore = (message: string) => ({
   type: DISPLAY_ERROR_ALERT,
   data: { message }
 })
+export const displayErrorAlert = (message: string) => (dispatch: Dispatch) => {
+  if (message === 'Invalid OTP token') {
+    Alert.alert(s.strings.otp_out_of_sync_title, s.strings.otp_out_of_sync_body)
+    return
+  }
+  dispatch(actions.dispatchActionObject(DISPLAY_ERROR_ALERT, { message }))
```

Comment:
yes. It needed to be a thunk in order to filter this error, and create a more human readable one per the task. 
So in making it a thunk, it broke the one call to the action creator in the `src/lib/ErrorAlert.js`. So I created a new action creator `displayErrorAlertStore` so that the lib would not be broken, and I could successfully filter error messages thrown with displayErrorAlert.